semantic-python/src/Language/Python/Core.hs

```diff
@@ -71,11 +72,11 @@ class Compile (py :: * -> *) where
              , MonadFail m
              )
           => py Span
-          -> m (t Name)
-          -> m (t Name)
+          -> (t Name -> m (t Name))
+          -> (t Name -> m (t Name))
```

Comment:
I wonder if `ContT` would improve the ergonomics of this interface.
Also, should the continuation be `a -> m a`?

Reply:
`ContT` would probably make it difficult for us to actually swap out the continuation, no? As with the various `alloca` and `with…` APIs in the `Foreign.*` modules, we need to be able to address the actions inside and following a scope separately.